src/l500/l500-depth.cpp

```diff
@@ -354,6 +355,44 @@ namespace librealsense
         AC_LOG( INFO, "Depth sensor calibration has been reset" );
     }
 
+    float l500_depth_sensor::get_max_usable_range() const
+    {
+        if( !supports_option( RS2_OPTION_MAX_USABLE_RANGE ) )
+            throw librealsense::wrong_api_call_sequence_exception( "max usable range option is not supported" );
+
+        if( get_option( RS2_OPTION_MAX_USABLE_RANGE ).query() != 1.0f )
+            throw librealsense::wrong_api_call_sequence_exception( "max usable range option is not on" );
+
+        if( ! is_streaming() )
+        {
+            throw librealsense::wrong_api_call_sequence_exception("depth sensor is not streaming!");
+        }
+
+       algo::max_range::max_usable_range mur;
+       float nest = static_cast<float>(_owner->get_temperatures().nest_avg);
+
+       return mur.get_max_range(nest);
+    }
+
+    bool l500_depth_sensor::is_max_range_preset() const
+    {
+        auto res = _owner->_hw_monitor->send(command(ivcam2::IRB, 0x6C, 0x2, 0x1));
```

Comment:
Please comment:
`// We want to disable max-usable-range when not in a particular preset:`

Reply:
👍